.github/workflows/automated-tests.yml

```diff
@@ -0,0 +1,127 @@
+name: Automated Tests
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  automated-tests:
+    name: ${{ matrix.tests.command }}
+    runs-on: ubuntu-latest
+
+    env:
+      CI: true
+      RAILS_ENV: test
+      TEST_GROUP: ${{ matrix.tests.command }}
+      CUCUMBER_RETRY: 1
+      CUCUMBER_FORMAT: Ao3Cucumber::Formatter
+
+    services:
+      database:
+        image: mariadb:10.5.4-focal
+        env:
+          MYSQL_ROOT_PASSWORD: password
+          MYSQL_DATABASE: otwarchive_test
+        ports:
+          - 3306:3306
+
+      elasticsearch:
+        image: elasticsearch:6.8.9
+        ports:
+          - 9200:9200
+
+      memcached:
+        image: memcached:1.5
+        ports:
+          - 11211:11211
+
+    strategy:
+      max-parallel: 5
+      matrix:
+        tests:
+          - command: rake db:otwseed
+            cucumber: 'false'
+          - command: rspec spec/controllers
+            cucumber: 'false'
+          - command: rspec spec/models
+            cucumber: 'false'
+          - command: rspec --exclude-pattern 'spec/{controllers,models}/**/*.rb'
+            cucumber: 'false'
+          - command: cucumber features/admins
+          - command: cucumber features/bookmarks
+          - command: cucumber features/collections
+          - command: cucumber features/comments_and_kudos
+          - command: cucumber features/gift_exchanges
+          - command: cucumber features/importing
+          - command: cucumber features/other_a
+          - command: cucumber features/other_b
+          - command: cucumber features/prompt_memes_a
+          - command: cucumber features/prompt_memes_b
+          - command: cucumber features/prompt_memes_c
+          - command: cucumber features/search
+          - command: cucumber features/tag_sets
+          - command: cucumber features/tags_and_wrangling
+          - command: cucumber features/users
+          - command: cucumber features/works
+            ebook: 'true'
+
+    steps:
```

Comment:
Actions were disabled in this repo. I just changed the setting to "Allow select actions", with "Allow actions created by GitHub" and "Allow Marketplace actions by verified creators" checked, for the record.